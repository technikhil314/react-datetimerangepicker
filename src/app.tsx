import React from 'react';
import PropTypes from 'prop-types';

interface IProps {
  name: string;
}

export const App: React.FunctionComponent<IProps> = function ({ name }) {
  return <h1 data-testid="app">Hello {name}</h1>;
};

App.defaultProps = {
  name: 'World',
};

App.propTypes = {
  name: PropTypes.string.isRequired,
};
