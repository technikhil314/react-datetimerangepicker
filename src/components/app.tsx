/** @jsx jsx */
import { css, jsx } from '@emotion/core';
import PropTypes from 'prop-types';
import React from 'react';
import phoneImg from '../../public/assets/icons/phone-receiver.png';
import bigImg from '../../public/assets//images/xps-DYAf-8UTFN8-unsplash.jpg';
import anotherBigImg from '../../public/assets//images/xps-DYAf-8UTFN8-unsplash.jpeg';
export interface AppProps {
  name: string;
}

export const App: React.FunctionComponent<AppProps> = function () {
  return (
    <div data-testid="app">
      <div
        css={css`
          padding: 32px;
          background-color: hotpink;
          font-size: 24px;
          border-radius: 4px;
        `}
      >
        {[1, 2, 3, 4, 5].map((x) => (
          <h1
            css={css`
              &:hover {
                color: var(--demo, #fff);
              }
            `}
            key={x}
          >
            {x}
          </h1>
        ))}
        <img src={bigImg} alt=""></img>
        <img src={anotherBigImg} alt=""></img>
        <img src={phoneImg} alt=""></img>
      </div>
    </div>
  );
};

App.defaultProps = {
  name: 'World',
};

App.propTypes = {
  name: PropTypes.string.isRequired,
};
