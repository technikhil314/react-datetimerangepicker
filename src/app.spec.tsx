import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { App } from './app';

test('shows the children when the checkbox is checked', () => {
  const { container } = render(<App name="demo"></App>);
  expect(screen.getByTestId('app')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
