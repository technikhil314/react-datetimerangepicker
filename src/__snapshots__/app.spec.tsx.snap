// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows the children when the checkbox is checked 1`] = `
.emotion-5 {
  padding: 32px;
  background-color: hotpink;
  font-size: 24px;
  border-radius: 4px;
}

.emotion-0:hover {
  color: var(--demo,#fff);
}

<div>
  <div
    data-testid="app"
  >
    <div
      class="emotion-5"
    >
      <h1
        class="emotion-0"
      >
        1
      </h1>
      <h1
        class="emotion-0"
      >
        2
      </h1>
      <h1
        class="emotion-0"
      >
        3
      </h1>
      <h1
        class="emotion-0"
      >
        4
      </h1>
      <h1
        class="emotion-0"
      >
        5
      </h1>
    </div>
    <img
      src=""
    />
    <img
      src=""
    />
    <img
      src=""
    />
  </div>
</div>
`;
